spec.html

```diff
@@ -32162,26 +32225,122 @@ <h1>
+          1. Let _result_ be a new empty List.
+          1. For each element _identifier_ of _identifiers_, do
+            1. Let _primary_ be _identifier_.
+            1. If _identifier_ is a non-primary time zone identifier and _identifier_ is not *"UTC"*, then
+              1. Set _primary_ to the name of the primary time zone identifier that _identifier_ resolves to, according to the rules for resolving Link names in the IANA Time Zone Database.
+              1. NOTE: If _identifier_ resolves to another non-primary time zone identifier, then the implementation must continue resolution of the entire chain until its terminal primary time zone identifier.
+            1. If _primary_ is one of *"Etc/UTC"*, *"Etc/GMT"*, or *"GMT"*, set _primary_ to *"UTC"*.
+            1. Let _record_ be the Time Zone Identifier Record { [[Identifier]]: _identifier_, [[PrimaryIdentifier]]: _primary_ }.
+            1. Append _record_ to the end of _result_.
+          1. Assert: One element of _result_ is the Time Zone Identifier Record { [[Identifier]]: *"UTC"*, [[PrimaryIdentifier]]: *"UTC"* }.
+          1. Return _result_.
+        </emu-alg>
+
+        <emu-note>
+          Time zone identifiers in the IANA Time Zone Database can change over time.
+          At a minimum, it is recommended that implementations limit changes to the result of AvailableTimeZoneIdentifiers to the changes allowed by GetAvailableTimeZoneIdentifier, for the lifetime of the surrounding agent.
+          Due to the complexity of supporting these requirements, it is recommended that the result of AvailableTimeZoneIdentifiers remains the same for the lifetime of the surrounding agent.
+        </emu-note>
+      </emu-clause>
+
+      <emu-clause id="sec-hosttimezoneidentifier" oldids="sec-defaulttimezone" type="implementation-defined abstract operation">
+        <h1>HostTimeZoneIdentifier ( ): a String</h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>
+            It returns a String value representing the host environment's current time zone, which is either a String representing a UTC offset for which IsTimeZoneOffsetString returns *true*, or a primary time zone identifier.
+          </dd>
+        </dl>
 
         <emu-alg>
-          1. Return *"UTC"*.
+          1. If the implementation only supports the UTC time zone, return *"UTC"*.
+          1. Let _hostTimeZoneString_ be the String representing the host environment's current time zone, either a time zone identifier or a UTC offset string.
+          1. If IsTimeZoneOffsetString(_hostTimeZoneString_) is *true*, return _hostTimeZoneString_.
+          1. Assert: GetAvailableTimeZoneIdentifier(_hostTimeZoneString_) is not ~empty~.
+          1. Assert: GetAvailableTimeZoneIdentifier(_hostTimeZoneString_).[[Identifier]] is GetAvailableTimeZoneIdentifier(_hostTimeZoneString_).[[PrimaryIdentifier]].
+          1. Return _hostTimeZoneString_.
```

Comment:
Suggestion: require output to use correct letter casing.
```suggestion
          1. Let _timeZoneRecord_ be GetAvailableTimeZoneIdentifier(_hostTimeZoneString_).
          1. Assert: _timeZoneRecord_ is not ~empty~.
          1. Assert: _timeZoneRecord_.[[Identifier]] is _timeZoneRecord_.[[PrimaryIdentifier]].
          1. Return _timeZoneRecord_.[[PrimaryIdentifier]].
```